### DISCUSSION TITLE
Battery Outputs

### DISCUSSION THREAD

**Discussion by miklos73** [tolwi/hassio-ecoflow-cloud#152] [Ideas]
Hi!
The integration of Delta Pro and Powersteam worked well. (I don't see the second additional battery/smart battery, but that doesn't matter)
What is more painfull is that I can't read the amount of energy taken from the batteries either via Powerstream or Delta Pro. (Wh or kWh)
So I see in the Energy Monitor that xy kWh were charged into the batteries, but (always) 0 kWh was taken out.
Is it possible to integrate this?
Greetings from Switzerland

**Comment by tolwi** [comment]
you can use [Integration - Riemann sum integral](https://www.home-assistant.io/integrations/integration/?#energy)

**Reply by miklos73** [reply]
Hey tolwi!
Thank you very much for your help!
It worked so far, I now have a continuously counting and totaled value for the amount of energy taken from the EcoFlow battery
Is it possible to create a second value that only shows the daily amount (from 00:00 to 24:00)?
So it is set to zero every day at midnight?

**Reply by tolwi** [reply]
sure - [utility_meter](https://www.home-assistant.io/integrations/utility_meter/)

**Reply by Fyr42** [reply]
You can do an helper based on this sensor, with a "utility meter" and reset/cycle it every period at your convenience. 
<img width="601" alt="Screenshot 2023-09-26 at 15 09 57" src="https://github.com/tolwi/hassio-ecoflow-cloud/assets/119723794/85637e4f-8935-4539-87a4-d645f58ab05f">

**Reply by miklos73** [reply]
Thank you both, it works as expected!
(sorry, I'm absolutely new to home assistant too)

**Reply by Expalmi** [reply]
@miklos73 Hi, I do have the same problem, can you please explain what entity you choose to convert with riemann integration?

**Reply by Fyr42** [reply]
Riemann integration takes watt to convert to watt per hour. You choose any entity in watt.
Like powerstream_battery_input_watts , powerstream_inverter_output_watts ,

**Reply by Expalmi** [reply]
Thanks a lot.
